homeassistant/components/nextbus/__init__.py

```diff
@@ -4,15 +4,41 @@
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
 
+from .const import CONF_AGENCY, CONF_ROUTE, CONF_STOP, DOMAIN
+from .coordinator import NextBusDataUpdateCoordinator
+
 PLATFORMS = [Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up platforms for NextBus."""
+    entry_agency = entry.data[CONF_AGENCY]
+
+    coordinator: NextBusDataUpdateCoordinator = hass.data.setdefault(DOMAIN, {}).get(
+        entry_agency
+    )
+    if coordinator is None:
+        coordinator = NextBusDataUpdateCoordinator(hass, entry_agency)
+        hass.data[DOMAIN][entry_agency] = coordinator
```

Comment:
Don't store the coordinator in `hass.data` until after the first refresh which may raise `ConfigEntryNotReady`.